src/main/java/org/springframework/integration/aws/outbound/KplMessageHandler.java

```diff
@@ -115,8 +124,60 @@ public void setConverter(Converter<Object, byte[]> converter) {
 		setMessageConverter(new ConvertingFromMessageConverter(converter));
 	}
 
+	/**
+	 * When in KPL mode, the setting allows handling backpressure on the KPL native process.
+	 * Setting this value would enable a sleep on the KPL Thread for the specified number of milliseconds defined in
+	 * maxRecordInFlightsSleepDurationInMillis.
+	 *
+	 * @param maxRecordsInFlight Defaulted to 0. Value of 0 indicates that Backpressure handling is not enabled.
+	 * Specify a positive value to enable back pressure.
+	 * @since 3.0.9
+	 */
+	public void setMaxRecordsInFlight(long maxRecordsInFlight) {
+		Assert.isTrue(maxRecordsInFlight > 0, "'maxRecordsInFlight must be greater than 0.");
+		this.maxInFlightRecords = maxRecordsInFlight;
+	}
+
+	/**
+	 * The setting allows handling backpressure on the KPL native process.
+	 * Enabled when maxOutstandingRecordsCount is greater than 0.
+	 * The configurations puts the KPL Thread to sleep for the specified number of milliseconds.
+	 *
+	 * @param maxInFlightRecordsBackoffDuration Default is 100ms.
+	 * @since 3.0.9
+	 */
+	public void setMaxInFlightRecordsBackoffDuration(int maxInFlightRecordsBackoffDuration) {
+		Assert.isTrue(maxInFlightRecordsBackoffDuration > 0,
+				"'maxRecordInFlightsSleepDurationInMillis must be greater than 0.");
+		this.maxInFlightRecordsBackoffDuration = maxInFlightRecordsBackoffDuration;
+	}
+
+	/**
+	 * The setting allows handling backpressure on the KPL native process using exponential retry.
+	 *
+	 * @param maxInFlightRecordsBackoffRate The property enables a back off rate to
+	 * 	 * define the exponential retry duration defined in setMaxInFlightRecordsBackoffDuration. Default is 2
+	 * @since 3.0.9
+	 */
+	public void setMaxInFlightRecordsBackoffRate(int maxInFlightRecordsBackoffRate) {
+		this.maxInFlightRecordsBackoffRate = maxInFlightRecordsBackoffRate;
+	}
+
+	/**
+	 * The setting allows handling backpressure on the KPL native process using exponential retry. On attempts
```

Comment:
Probably the javadoc is causing a confusion as per your comment.  I have removed the mentions "KPL Native Process" from the java docs. 
Please let me know if you think otherwise or my understanding is wrong.